### PR TITLE
KafkaSource/KafkaBinding v1beta1 e2e

### LINKED ISSUE #1375: KafkaSource/KafkaBinding update e2e tests to use v1beta1
**Problem**
Part of https://github.com/knative/eventing-contrib/issues/1332

Update e2e tests to use new version

Example: https://github.com/knative/eventing/pull/3430

### PR DESCRIPTION
**Pull request by aliok**
Fixes #1375

## Proposed Changes

- E2e tests for KafkaSource and KafkaBinding v1beta1

### CODE REVIEW

**Comment by aliok** [comment]
cc @nlopezgi @vaikas @lionelvillard

File: test/e2e/kafka_binding_test.go [hunk 1]
```diff
@@ -47,28 +47,54 @@ func testKafkaBinding(t *testing.T, messageKey string, messageHeaders map[string
...
-			APIVersion: "batch/v1",
-			Kind:       "Job",
-			Selector: &metav1.LabelSelector{
-				MatchLabels: selector,
+	t.Logf("Creating KafkaBinding %s", version)
+	switch version {
+	case "v1alpha1":
+		contribtestlib.CreateKafkaBindingV1Alpha1OrFail(client, contribresources.KafkaBindingV1Alpha1(
+			kafkaBootstrapUrl,
+			&tracker.Reference{
```
Comment (nlopezgi) [inline test/e2e/kafka_binding_test.go:76]: Is this object identical to the one declared in line 87? If so, consider declaring it before the switch and reusing it?

File: test/e2e/kafka_binding_test.go [hunk 2]
```diff
@@ -116,8 +142,11 @@ func TestKafkaBinding(t *testing.T) {
 	}
 	for name, tc := range tests {
 		tc := tc
-		t.Run(name, func(t *testing.T) {
-			testKafkaBinding(t, tc.messageKey, tc.messageHeaders, tc.messagePayload, tc.expectedData)
+		t.Run(name+"-v1alpha1", func(t *testing.T) {
```
Comment (nlopezgi) [inline test/e2e/kafka_binding_test.go:145]: Would it be worth the effort to have some struct with the versions to test and iterate over those here? (probably only worth if more versions need to be maintained than the current two in the future, not sure about how this is expected to evolve)

Comment (vaikas) [reply test/e2e/kafka_binding_test.go:145]: How about we create an issue to track this and do as a follow up?

**Comment by vaikas** [comment]
/lgtm
/approve

/hold
if you want to address the comments, feel free to remove the hold.

File: test/e2e/kafka_binding_test.go (hunk 2 above)
Comment (aliok) [reply test/e2e/kafka_binding_test.go:145]: good suggestion @nlopezgi.
things here can be extracted into a helper function and then the test functions can call the helpers.
if you don't mind, let's do that afterwards or maybe even later when we have other versions.

**Pull request merged**